src/coordinate_transformations.jl

```diff
@@ -0,0 +1,91 @@
+mutable struct CRS2CRS <: CoordinateTransformations.Transformation
+    rep::Ptr{Cvoid}
+    direction::PJ_DIRECTION
+end
+
+function CRS2CRS(source::String, target::String, direction::PJ_DIRECTION = PJ_FWD; area = C_NULL, normalize = true)
+
+    if direction == PJ_IDENT || source == direction
+        return IdentityTransformation()
+    end
+
+    pj_init = proj_create_crs_to_crs(source, target, area)
+
+    obj = if normalize
+        pj_normalized = proj_normalize_for_visualization(pj_init)
+        proj_destroy(pj_init)
+        CRS2CRS(pj_normalized, direction)
+    else
+        CRS2CRS(pj_normalized, direction)
+    end
+
+    finalizer(obj) do obj
+        proj_destroy(obj.rep)
+    end
+
+    return obj
+end
+
+function CRS2CRS(source::Projection, target::Projection, direction::PJ_DIRECTION = PJ_FWD; area = C_NULL, normalize = true)
+    return CRS2CRS(
+        sprint(print, source),
+        sprint(print, target),
+        direction;
+        area = area,
+        normalize = true
+    )
+end
+
+Base.inv(cs::CRS2CRS) = CRS2CRS(cs.rep, inv(cs.direction))
+
+function Base.inv(dir::PJ_DIRECTION)
+    return if dir == PJ_IDENT
+        PJ_IDENT
+    elseif dir == PJ_FWD
+        PJ_INV
+    else
+        PJ_FWD
+    end
+end
+
+function CoordinateTransformations.transform_deriv(cs::CRS2CRS, x)
+    coord = proj_coord(x...)
+    factors = proj_factors(cs.rep, coord)
+    return [dx_dlam dx_dphi; dy_dlam dy_dphi]
+end
+
+function (transformation::CRS2CRS)(x::T) where T
```

Comment:
Is this `where T` needed?  Seems it could be removed.

Reply:
Would the method still specialize on types with size information (e.g. static arrays) then?

Reply:
Ugh, [that trick](https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Be-aware-of-when-Julia-avoids-specializing-1) again! According to the docs it's only important for inputs of type `Function`, `Type` and `Vararg`. So I guess not needed here; I'd delete it, unless you can measure a performance difference.